src/features/auth/login/PasswordForm.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { Redirect, useHistory, useLocation } from "react-router-dom";
+import { loginRoute } from "../../../AppRoutes";
+import Button from "../../../components/Button";
+import TextInput from "../../../components/TextInput";
+import { useAppDispatch, useTypedSelector } from "../../../store";
+import { passwordLogin } from "../authActions";
+import { clearError } from "../authSlice";
+
+export default function PasswordForm() {
+  const dispatch = useAppDispatch();
+  const authLoading = useTypedSelector((state) => state.auth.loading);
+
+  const { handleSubmit, register } = useForm<{ password?: string }>();
+  const history = useHistory();
+  const location = useLocation<{ username?: string }>();
```

Comment:
```suggestion
  const location = useLocation<{ username: string }>();
```
I think you can type `username` as always present since you redirect the user back to fill that in in line 36 if it's missing?